modules/ctl/src/main/scala/ioref.scala

```diff
@@ -0,0 +1,17 @@
+// Copyright (c) 2016-2017 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package gem.ctl
+
+import cats.effect.IO
+
+// N.B. this isn't in cats-effect yet
+@SuppressWarnings(Array("org.wartremover.warts.Var"))
+final class IORef[A] private (var value: A) {
+  def mod(f: A => A): IO[Unit] = IO(value = f(value))
+  def read: IO[A] = IO(value)
+}
+object IORef {
+  def apply[A](a: A): IO[IORef[A]] =
+    IO(new IORef(a))
+}
```

Comment:
Are there any plans to have something that fulfills the role of `IORef` in `cats-effect`? This can come handy for the seqexec, too, but I was looking  into using `fs2` [`Ref`](https://oss.sonatype.org/service/local/repositories/releases/archive/co/fs2/fs2-core_2.12/0.9.6/fs2-core_2.12-0.9.6-javadoc.jar/!/fs2/util/Async$$Ref.html).